src/java/org/apache/cassandra/cql3/statements/schema/AlterSchemaStatement.java

```diff
@@ -46,7 +47,10 @@ protected AlterSchemaStatement(String keyspaceName)
 
     public void validate(ClientState state)
     {
-        // no-op; validation is performed while executing the statement, in apply()
+        // validation is performed while executing the statement, in apply()
+
+        // Cache our ClientState for use by guardrails
+        this.state = state;
```

Comment:
this pattern keeps popping up... wondering if we should "fix" our apis to always provide this?